dev/doxygen/pages/release_workflow.md

```diff
@@ -0,0 +1,155 @@
+Whether the file format is stable or not is controlled by the define
+`FILE_FORMAT_STABLE`. See following chapters for details.
+
+
+# Git Branches {#doc_release_workflow_branches}
+
+The continuous development happens on feature/bugfix branches (or pull requests),
+which are merged into the `master` branch as soon as they are finished. As new
+features sometimes also introduce file format changes, the `master` branch must
+always be considered to contain the (unstable) *next* major version of the
+application. For example if the last stable release was version "2.3.4", the
+`master` must immediately be considered as version "3.0.0-unstable".
+
+This also means that on the `master` branch we cannot provide updates for
+already released versions. So we have to provide such updates in release
+branches. For every major version we create a release branch, which is then used
+for *any* update for that major version. For example on the branch `release/2`
+we would provide the releases "2.0", "2.0.1" (bugfix) and "2.1.0" (new feature).
+
+![Git Branches](release_workflow_diagram.png)
+
+Features and bugfixes which do not affect the file format may be cherry-picked
+from `master` to release branches. This way we can publish updates for the last
+officially released application version while the next major version is still in
+development. Generally we do *not* maintain older releases than the last stable
+release because this would be too much effort. So for example as soon as the
+version "0.2.0" is released, the version "0.1.x" is no longer maintained.
+
+
+# Git Tags
+
+For every officially released version we create a Git tag on the corresponding
+release branch. The tag name contains all three numbers of the application's
+version number. Examples: "0.1.0-rc1", "0.1.0", "1.2.0", "1.2.1"
+
+
+# Warning in Unstable Versions {#doc_release_workflow_warning}
+
+Because (at least) the `master` branch is always unstable, it's very dangerous
+for end users to work productively with this version (it could break projects
+etc.!). So we should warn users in that situation.
+
+This is done by displaying warnings in the application's GUI. Their visibility
+is controlled by the preprocessor define `FILE_FORMAT_STABLE`. It's set to `0`
+(warning visible) on all branches, except on release branches where it's set to
+`1` (warning disabled). As release branches always have a stable file format, it
+should be safe to use *any* state of a release branch (even if not officially
+released).
+
+
+# Translations {#doc_release_workflow_translations}
+
+We use [Transifex] to translate LibrePCB into other languages. As Transifex
+doesn't have support for branches, we use a separate [resource] for each major
+application version. The translations for the `master` branch are contained in
+the resource `librepcb.ts`, while translations for releases are contained in
+resources like `librepcb-0.1.ts`.
```

Comment:
Does Transifex sync translations between the different resources? So if someone translates strings in librepcb-0.1.ts, does that also affect `librepcb.ts`?

Reply:
I guess the [Translation Memory](https://docs.transifex.com/setup/translation-memory/) should automatically fill in translated strings from any other resource with the same context.

Generally I would propose that translators always translate `librepcb.ts`, not resources of older releases. Translations of older releases should be updated only with the translation memory. But if the translation memory works well, maybe it even doesn't care on what resource translators are working on :)